doc/source/developing/creating_derived_fields.rst

```diff
@@ -354,3 +354,79 @@ And you wanted to debug it, you could do:
 
 And now, when that derived field is actually used, you will be placed into a
 debugger.
+
+Deprecating Derived Fields
+--------------------------
+
+Occasionally, one may want to deprecate a derived field in yt, normally
+because naming conventions for fields have changed, or simply because a
+field has outlived its usefulness. There are two ways to mark fields as
+deprecated in yt.
+
+The first way is if you simply want to mark a specific derived field as
+deprecated. In that case, you call
+:meth:`~yt.fields.field_info_container.FieldInfoContainer.add_deprecated_field``:
+
+.. code-block:: python
+
+    def _cylindrical_radial_absolute(field, data):
+        """This field is deprecated and will be removed in a future version"""
+        return np.abs(data[ftype, f"{basename}_cylindrical_radius"])
+
+
+    registry.add_deprecated_field(
+        (ftype, f"cylindrical_radial_{basename}_absolute"),
+        sampling_type="local",
+        function=_cylindrical_radial_absolute,
+        since="4.0",
+        removal="4.1",
+        units=field_units,
+        validators=[ValidateParameter("normal")],
+    )
+
+Note that the signature for
+:meth:`~yt.fields.field_info_container.FieldInfoContainer.add_deprecated_field``
+is the same as :meth:`~yt.fields.field_info_container.FieldInfoContainer.add_field``,
+with the exception of the ``since`` and ``removal`` arguments which indicate in
+what version the field was deprecated and in what version it will be removed.
+The effect is to add a warning to the logger when the field is first used:
+
+.. code-block:: python
+
+    import yt
+
+    ds = yt.load("GasSloshing/sloshing_nomag2_hdf5_plt_cnt_0100")
+    sp = ds.sphere("c", (100.0, "kpc"))
+    print(sp["gas", "cylindrical_radial_velocity_absolute"])
+
+.. code-block:: pycon
+
+    yt : [WARNING  ] 2021-03-09 16:30:47,460 The Derived Field
+    ('gas', 'cylindrical_radial_velocity_absolute') is deprecated
+    as of yt v4.0 and will be removed in yt v4.1.
+
+The second way to deprecate a derived field is to take an existing field
+definition and change its name. In order to mark the original name as deprecated,
+use the :meth:`~yt.fields.field_info_container.FieldInfoContainer.alias`` method
+and pass the ``since`` and ``removal`` arguments (see above) as a tuple in the
+``deprecate`` keyword argument:
+
+.. code-block:: python
+
+    registry.alias(
+        (ftype, "kinetic_energy"),
+        (ftype, "kinetic_energy_density"),
+        deprecate=("4.0", "4.1"),
+    )
+
+In this case, the log message reports to the user what field they should use:
```

Comment:
Sorry, I thought you said that we needed to use the logger to avoid doing things twice?

Reply:
This might be a leftover from a first draft of my review. I went back and corrected some stuff when I understood why you couldn't use issue_deprecation_warning, but I likely forgot about this one. You can resolve !